src/base/searchengine.h

```diff
@@ -38,10 +40,133 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
+{
+public:
+    typedef T ComponentType;
+    typedef Version<T,N> this_type;
+
+    Version()
+    {
+        m_components.fill(T());
+    }
+
+    template <class ...Other>
+    Version(Other... components)
+        : m_components {{components...}}
+    {
+    }
+
+    Version(const QString &version)
```

Comment:
>It is weird to have a ctor takingQString as parameter

The essence of `Version` class is just a tuple of numbers, how could someone imagine he could construct a tuple of numbers via a QString?

So it would be more explicit to do something like:
```c++
class Version
{
    static Version fromQString(const QString &s) {/* parsing work */}
    static Version fromQByteArray(const QByteArray &s) {/* parsing work */}
}

Version ver = Version::fromQString(QString("1.2.3"));
```